### ISSUE TITLE
nn.functional.kl_div(p, p) is not 0

### ISSUE THREAD

**Issue by areesejimmy** [pytorch/pytorch#51716]
## 🐛 Bug

The **Kullback-Leibler** divergence between a distribution `p` and itself must be equal to `0`. Using `nn.functional.kl_div(p, p)` does not output `0`.

## To Reproduce

```
import torch
import torch.nn as nn
n = 16
d = 16
a = torch.rand((n, d))
p = nn.functional.softmax(a, dim=-1)
nn.functional.kl_div(p, p)
```

## Expected behavior

`tensor(0.)`

## Actual behavior

`tensor(-0.1753)`
Changing the dimensions of the input changes the output but it's not zero.

## Environment
```

PyTorch version: 1.7.1
Is debug build: False
CUDA used to build PyTorch: 10.1
ROCM used to build PyTorch: N/A

OS: Ubuntu 16.04.4 LTS (x86_64)
GCC version: (Ubuntu 5.4.0-6ubuntu1~16.04.10) 5.4.0 20160609
Clang version: Could not collect
CMake version: version 3.5.1
```

**Comment by SsnL** [comment]
https://pytorch.org/docs/stable/nn.functional.html#torch.nn.functional.kl_div

> log_target (bool) – A flag indicating whether target is passed in the log space. It is recommended to pass certain distributions (like softmax) in the log space to avoid numerical issues caused by explicit log. Default: False

**Comment by chl19940220** [comment]
```nn.functional.kl_div(p, p, log_target=True)``` will get 0
